### PR TITLE
Redis repository

### PR DESCRIPTION
**Pull request by VolodymyrM27**

### CODE REVIEW

File: compose.yaml [hunk 1]
```diff
@@ -59,6 +54,11 @@ services:
     environment:
       SCHEMA_REGISTRY_HOST_NAME: schema-registry
       SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS: PLAINTEXT://kafka1:29092
+
+  redis-air-guardian:
+    image: redis:alpine3.18
+    ports:
+      - "6379:6379"
```
Comment (ajax-gnatiuk-s) [inline compose.yaml:61]:
```suggestion
      - "6379:6379"
      
```

File: airguardian-core/src/test/kotlin/systems/ajax/motrechko/airguardian/controller/nats/GetAllDronesNatsControllerTest.kt [hunk 2]
```diff
@@ -33,13 +34,18 @@ class GetAllDronesNatsControllerTest {
     @Autowired
     private lateinit var droneRepository: DroneRepository
 
+    @Autowired
+    private lateinit var reactiveRedisRepository: ReactiveRedisTemplate<String,Drone>
+
     @AfterEach
-    fun cleanDB() {
+    fun cleanAAfterTest() {
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/src/test/kotlin/systems/ajax/motrechko/airguardian/controller/nats/GetAllDronesNatsControllerTest.kt:41]:
```suggestion
    fun cleanAfterTest() {
```

File: airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    override fun save(entity: Drone): Mono<Drone> =
+        droneRepository.save(entity)
+            .flatMap { savedDrone ->
+                reactiveRedisRepository.opsForValue()
+                    .set(
+                        dronePrefix + savedDrone.id.toHexString(),
+                        savedDrone,
+                        Duration.ofMinutes(ttlMinutes.toLong())
+                    )
+                    .then(savedDrone.toMono())
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt:32]:
```suggestion
                saveDroneToRedisCache(drone)
```

?

File: airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import java.time.Duration
+
+@Repository
+class DroneRedisRepository(
+    private val droneRepository: DroneRepository,
+    private val reactiveRedisRepository: ReactiveRedisTemplate<String, Drone>,
+    @Value("\${spring.data.redis.key.prefix}") private val dronePrefix: String,
+    @Value("\${spring.data.redis.ttl.minutes}") private val ttlMinutes: String
+) : DroneCacheableRepository {
+    override fun save(entity: Drone): Mono<Drone> =
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt:23]:
```suggestion
) : DroneCacheableRepository {

    override fun save(entity: Drone): Mono<Drone> =
```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

File: airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        droneRepository.save(entity)
+            .flatMap { savedDrone ->
+                reactiveRedisRepository.opsForValue()
+                    .set(
+                        dronePrefix + savedDrone.id.toHexString(),
+                        savedDrone,
+                        Duration.ofMinutes(ttlMinutes.toLong())
+                    )
+                    .thenReturn(savedDrone)
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt:32]:
```suggestion
                saveDroneToRedisCache(savedDrone)
```
?

Comment (ajax-gnatiuk-s) [reply airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt:32]: Note! I've written the [same comment](https://github.com/VolodymyrM27/AirGuardian/pull/13#discussion_r1382641555) earlier

Comment (VolodymyrM27) [reply airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt]: I didn't understand the first time, my bad

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**